include/zenoh-pico/collections/pointer.h

```diff
@@ -19,19 +19,25 @@
 #include <stdint.h>
 
 #if ZENOH_C_STANDARD != 99
+#ifndef __cplusplus
 #include <stdatomic.h>
+#else
+#include <atomic>
+#define _Atomic(X) std::atomic<X>
+using namespace std;
+#endif
 
 /*------------------ Internal Array Macros ------------------*/
 #define _Z_POINTER_DEFINE(name, type)                                                           \
     typedef struct {                                                                            \
         type##_t *ptr;                                                                          \
-        atomic_uint *_cnt;                                                                      \
+        _Atomic unsigned int *_cnt;                                                             \
```

Comment:
This looks odd to me: without the parentheses, is this interpreted as `_Atomic(unsigned int *)_cnt`, `_Atomic(unsigned int) *_cnt` or `_Atomic(unsigned) int *_cnt`?

Reply:
It is interpreted as `_Atomic(unsigned int) *_cnt`.

I would have preferred the more explicit and use `_Atomic(unsigned int) *_cnt` but Zephyr did not agree with me. I had to do this change in [cf6341c](https://github.com/eclipse-zenoh/zenoh-pico/pull/164/commits/cf6341ce084de4cc24edc63673977a80fd469e45) to make all the ports happy.

Reply:
I think
```c++
#ifndef __cplusplus
#include <stdatomic.h>
#define z_atomic(X) _Atomic X
#else
#include <atomic>
#define z_atomic(X) std::atomic<X>
using namespace std;
#endif
```
would solve the Zephyr issue, and make which part is atomic clearer.

Reply:
This is done.